assets/dev/js/editor/document/repeater/commands/remove.js

```diff
@@ -58,11 +58,16 @@ export class Remove extends $e.modules.editor.document.CommandHistoryBase {
 
 			// Remove from container and add to result.
 			result.push( repeaterContainer.children.splice( index, 1 ) );
-
 			collection.remove( model );
 
-			// Trigger render on widget but with the settings of the control.
-			repeaterContainer.render();
+			if ( 'Accordion' === container.label ) {
```

Comment:
@davseve This base command should be generic and not aware of any specific feature.

Change this if statement to:

```
if ( isWidgetSupportNesting( widget_name ) && isWidgetSupportImprovedRepeaters( widget_name ) ) {
}
```

isWidgetSupportNesting() is already defined in `elementor/modules/nested-elements/assets/js/editor/utils.js`

You can create a function `isWidgetSupportImprovedRepeaters()` and then add some property on the widget config like `support_improved_repeaters`

Whatever else you need like `.e-n-accordion-item` can also be a part of the widget config. `protected function get_initial_config() {}`

Reply:
@davseve This just manually removes the item title, so I just want to make sure the child Container is also removed?

Reply:
Thanks for the detailed explanation, I'll do so.
And yes, the remove removes the entire element and its content.

Reply:
For `support_improved_repeaters => true`, we can maybe instead use `repeater_action_type => 'single-node'`.
I will leave it up to you to put some thought into this.

Reply:
@ArmanPurtoian Same for your task 🙏

Reply:
@davseve Lastly, you need to think about how we can support the different nested structures. In your case we were lucky that both items (title & container) were wrapped in one element that can be removed. But in Mega Menu the Container is somewhere else (and will be removed by the above mentioned hook). I know the MM is a different widget, but I think we should support the different kind of structures now in this base already.

Reply:
Note: this should be behind an experiment.